spec/features/saml/saml_spec.rb

```diff
@@ -273,6 +279,240 @@
+
+        click_button 'Submit'
+
+        visit sp2_authnrequest
+        @sp2_asserted_session_index = response_xmldoc.assertion_statement_node['SessionIndex']
+        click_button 'Submit'
+      end
+
+      it 'deactivates each authenticated Identity and logs the user out' do
+        visit destroy_user_session_url
+
+        click_button 'Submit' # logout request for first SP
+        click_button 'Submit' # logout request for second SP
+
+        logout_user.identities.each do |ident|
+          expect(ident.last_authenticated_at).to be_nil
+        end
+
+        expect(logout_user.active_identities).to be_empty
+
+        visit edit_user_registration_path
+        expect(page).to have_content I18n.t 'devise.failure.unauthenticated'
+      end
+
+      it 'references the correct SessionIndexes' do
+        visit destroy_user_session_url
+
+        expect(request_xmldoc.asserted_session_index).to eq(@sp2_asserted_session_index)
+        click_button 'Submit'
+
+        expect(request_xmldoc.asserted_session_index).to eq(@sp1_asserted_session_index)
+        click_button 'Submit'
+        visit edit_user_registration_path
+        expect(page).to have_content I18n.t 'devise.failure.unauthenticated'
+      end
+    end
+
+    context 'with multiple SP sessions and SP-initiated logout' do
+      let(:user) { create(:user, :signed_up) }
+      let(:response_xmldoc) { XmlDoc.new('feature', 'response_assertion') }
+      let(:request_xmldoc) { XmlDoc.new('feature', 'request_assertion') }
+
+      before do
+        visit sp1_authnrequest # sp1
+        authenticate_user(user)
+        @sp1_asserted_session_index = response_xmldoc.assertion_statement_node['SessionIndex']
+        click_button 'Submit'
+
+        visit sp2_authnrequest # sp2
+        @sp2_asserted_session_index = response_xmldoc.assertion_statement_node['SessionIndex']
+        click_button 'Submit'
+
+        request = OneLogin::RubySaml::Logoutrequest.new
+        settings = sp2_saml_settings # sp2
+        settings.name_identifier_value = user.uuid
+        visit request.create(settings)
+      end
+
+      it 'deactivates each Identity' do
+        expect(user.active_identities.size).to eq(2)
+        expect(page.current_path).to eq('/api/saml/logout')
+        click_button 'Submit' # LogoutRequest for first SP
+
+        # SP1 logs user out and responds with success
+        # User session is terminated at IdP and success
+        # is returned to SP2 (originating requestor)
+        click_button 'Submit' # LogoutResponse for originating SP
+
+        sp2 = ServiceProvider.new(sp2_saml_settings.issuer)
+
+        expect(page.current_url).
+          to eq(sp2.assertion_consumer_logout_service_url)
+        expect(user.active_identities.size).to eq(0)
+      end
+
+      it 'asserts correct SessionIndex' do
+        expect(request_xmldoc.asserted_session_index).
+          to eq(@sp1_asserted_session_index)
+        click_button 'Submit' # LogoutRequest for first SP
```

Comment:
Everything except the first line of this spec looks like it doesn't belong here since its already covered in the specs immediately above and below.   The name of spec seems to concur.